mssql/compiler.py

```diff
@@ -470,11 +480,43 @@ def as_sql(self):
 
         placeholder_rows, param_rows = self.assemble_as_sql(fields, value_rows)
 
-        if self.get_returned_fields() and self.connection.features.can_return_id_from_insert:
-            result.insert(0, 'SET NOCOUNT ON')
-            result.append((values_format + ';') % ', '.join(placeholder_rows[0]))
-            params = [param_rows[0]]
-            result.append('SELECT CAST(SCOPE_IDENTITY() AS bigint)')
+        if self.get_returned_fields() and self.can_return_columns_from_insert():
+            if self.can_return_rows_from_bulk_insert():
+                if not(self.query.fields):
+                    # There isn't really a single statement to bulk multiple DEFAULT VALUES insertions,
+                    # so we have to use a workaround:
+                    # https://dba.stackexchange.com/questions/254771/insert-multiple-rows-into-a-table-with-only-an-identity-column
+                    result = ["""
+                    WITH empty_rows AS (
+                        SELECT 1 AS n UNION ALL SELECT 1 AS n UNION ALL SELECT 1 AS n UNION ALL SELECT 1 AS n UNION ALL
+                        SELECT 1 AS n UNION ALL SELECT 1 AS n UNION ALL SELECT 1 AS n UNION ALL SELECT 1 AS n)
+                    MERGE INTO %s
+                    USING (SELECT TOP %s * FROM 
+                    (SELECT row_number() over (order by a.n) as number
```

Comment:
Doesn't it work with `SELECT 1 FROM` instead of `SELECT row_number() over (order by a.n) as number FROM` ?

Reply:
No it does not work with SELECT 1 FROM.

Reply:
Sorry, my suggestion was imprecise, but I've tried:
`SELECT 1 as _`
instead of 
`SELECT row_number() over (order by a.n) as number`
and it seems to work. Could help getting the trick as lean as possible.